File: frontend/src/types/dashboard-charts.ts
export interface MonthlyAmount {
  month: string;
  amount: string;
}

export interface InvoiceAgingBucket {
  status: string;
  count: number;
  total: string;
}

export interface OutstandingContact {
  contact_id: string;
  contact_name: string;
  outstanding_amount: string;
  invoice_count: number;
}

export interface OverviewMonth {
  month: string;
  income: string;
  expenses: string;
  cumulative_income: string;
  cumulative_expenses: string;
}

export interface OverviewData {
  year: number;
  months: OverviewMonth[];
  total_income: string;
  total_expenses: string;
  difference: string;
  available_years: number[];
}

export interface CashFlowItem {
  description: string;
  amount: string;
}

export interface CashFlowSection {
  label: string;
  inflows: string;
  outflows: string;
  net: string;
  items: CashFlowItem[];
}

export interface CashFlowReport {
  sections: CashFlowSection[];
  net_change: string;
  opening_balance: string;
  closing_balance: string;
  reconciliation_difference: string;
  from_date: string;
  to_date: string;
}

export interface AgingBucket {
  bucket: string;
  count: number;
  total: string;
}

export interface MonthlyCashFlow {
  month: string;
  inflows: string;
  outflows: string;
  net: string;
  cumulative_balance: string;
}

export interface CashFlowMonthlyReport {
  months: MonthlyCashFlow[];
  initial_balance: string;
  ending_balance: string;
  total_inflows: string;
  total_outflows: string;
  net_variation: string;
}

File: backend/crates/konto-api/src/dto/dashboard_charts.rs
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};
use utoipa::{IntoParams, ToSchema};

#[derive(Debug, Deserialize, IntoParams)]
pub struct MonthsParams {
    pub months: Option<u32>,
    pub format: Option<String>,
}

#[derive(Debug, Deserialize, IntoParams)]
pub struct LimitParams {
    pub limit: Option<u32>,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct MonthlyAmountResponse {
    pub month: String,
    #[schema(value_type = String)]
    pub amount: Decimal,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct InvoiceAgingBucketResponse {
    pub status: String,
    pub count: i64,
    #[schema(value_type = String)]
    pub total: Decimal,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct OutstandingContactResponse {
    pub contact_id: String,
    pub contact_name: String,
    #[schema(value_type = String)]
    pub outstanding_amount: Decimal,
    pub invoice_count: i64,
}

#[derive(Debug, Deserialize, IntoParams)]
pub struct OverviewParams {
    pub year: Option<i32>,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct OverviewMonthResponse {
    pub month: String,
    #[schema(value_type = String)]
    pub income: Decimal,
    #[schema(value_type = String)]
    pub expenses: Decimal,
    #[schema(value_type = String)]
    pub cumulative_income: Decimal,
    #[schema(value_type = String)]
    pub cumulative_expenses: Decimal,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct OverviewResponse {
    pub year: i32,
    pub months: Vec<OverviewMonthResponse>,
    #[schema(value_type = String)]
    pub total_income: Decimal,
    #[schema(value_type = String)]
    pub total_expenses: Decimal,
    #[schema(value_type = String)]
    pub difference: Decimal,
    pub available_years: Vec<i32>,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct CashFlowItemResponse {
    pub description: String,
    #[schema(value_type = String)]
    pub amount: Decimal,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct CashFlowSectionResponse {
    pub label: String,
    #[schema(value_type = String)]
    pub inflows: Decimal,
    #[schema(value_type = String)]
    pub outflows: Decimal,
    #[schema(value_type = String)]
    pub net: Decimal,
    pub items: Vec<CashFlowItemResponse>,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct CashFlowReportResponse {
    pub sections: Vec<CashFlowSectionResponse>,
    #[schema(value_type = String)]
    pub net_change: Decimal,
    #[schema(value_type = String)]
    pub opening_balance: Decimal,
    #[schema(value_type = String)]
    pub closing_balance: Decimal,
    #[schema(value_type = String)]
    pub reconciliation_difference: Decimal,
    pub from_date: String,
    pub to_date: String,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct AgingBucketResponse {
    pub bucket: String,
    pub count: i64,
    #[schema(value_type = String)]
    pub total: Decimal,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct MonthlyCashFlowResponse {
    pub month: String,
    #[schema(value_type = String)]
    pub inflows: Decimal,
    #[schema(value_type = String)]
    pub outflows: Decimal,
    #[schema(value_type = String)]
    pub net: Decimal,
    #[schema(value_type = String)]
    pub cumulative_balance: Decimal,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct CashFlowMonthlyReportResponse {
    pub months: Vec<MonthlyCashFlowResponse>,
    #[schema(value_type = String)]
    pub initial_balance: Decimal,
    #[schema(value_type = String)]
    pub ending_balance: Decimal,
    #[schema(value_type = String)]
    pub total_inflows: Decimal,
    #[schema(value_type = String)]
    pub total_outflows: Decimal,
    #[schema(value_type = String)]
    pub net_variation: Decimal,
}
